src/server/controllers/home-block.controller.ts

```diff
@@ -40,112 +46,30 @@ export const getHomeBlocksHandler = async ({
-
-                return {
-                  ...homeBlock,
-                  metadata,
-                  collection: {
-                    ...collection,
-                    items,
-                  },
-                };
-              }
-              case HomeBlockType.Leaderboard: {
-                if (!metadata.leaderboards) {
-                  return null;
-                }
-
-                const leaderboardIds = metadata.leaderboards.map((leaderboard) => leaderboard.id);
-
-                const leaderboardsWithResults = await getLeaderboardsWithResults({
-                  ids: leaderboardIds,
-                  isModerator: ctx.user?.isModerator || false,
-                });
-
-                return {
-                  ...homeBlock,
-                  metadata,
-                  leaderboards: leaderboardsWithResults.sort((a, b) => {
-                    if (!metadata.leaderboards) {
-                      return 0;
-                    }
-
-                    const aIndex =
-                      metadata.leaderboards.find((item) => item.id === a.id)?.index ?? 0;
-                    const bIndex =
-                      metadata.leaderboards.find((item) => item.id === b.id)?.index ?? 0;
-
-                    return aIndex - bIndex;
-                  }),
-                };
-              }
-              case HomeBlockType.Announcement: {
-                if (!metadata.announcements) {
-                  return null;
-                }
-
-                const announcementIds = metadata.announcements.ids;
-                const announcements = await getAnnouncements({
-                  ids: announcementIds,
-                  dismissed: input.dismissed,
-                  limit: metadata.announcements.limit,
-                  user: ctx.user,
-                });
-
-                if (!announcements.length) {
-                  // If the user cleared all announcements in home block, do not display this block.
-                  return null;
-                }
+    return homeBlocks;
+  } catch (error) {
+    throw throwDbError(error);
+  }
+};
 
-                return {
-                  ...homeBlock,
-                  metadata,
-                  announcements: announcements.sort((a, b) => {
-                    const aIndex = a.metadata.index ?? 999;
-                    const bIndex = b.metadata.index ?? 999;
+export const getHomeBlocksByIdHandler = async ({
+  ctx,
+  input,
+}: {
+  ctx: Context;
+  input: GetHomeBlockByIdInputSchema;
+}): Promise<HomeBlockWithData> => {
+  try {
+    const homeBlock = await getHomeBlockById({
+      ...input,
+      ctx,
+    });
 
-                    return aIndex - bIndex;
-                  }),
-                };
-              }
-              default:
-                return { ...homeBlock, metadata };
-            }
-          })
-          .filter(isDefined)
-      )
-    ).filter(isDefined);
+    if (!homeBlock) {
+      throw throwDbError('Home block not found');
```

Comment:
There's this handy helper method
```suggestion
      throw throwNotFoundError('Home block not found');
```